Update README with installation instructions

awvwgk opened the pull request:
Installation and integration instructions for fpm and meson.

certik reviewed (approved):
Looks great. Now that on Apple M1, when I install gfortran using conda-forge, the nested functions work in Debug mode, but in Release mode (`-O3`, etc.) the generated binary segfaults, if the nested function accesses a variable in the parent scope. So the test you added to Meson might not be sufficient, since it doesn't access a variable. I am not a fan of these tests, they seem to slow down the configure step. I think for fpm at least, we should just understand the platform and compiler versions, and *know* what works and what doesn't, no need to test for it like this (also what about cross-compilation when you cannot even run the binary..).

Anyway, this one is good to merge. The above are just some more general thoughts.

awvwgk commented:
Removed the check and added an option to disable the C-API. Our testsuite will verify whether a compiler fully supports our usage of nested functions or not.
